### ISSUE TITLE
Allow configurable file types to upload inside Child Directories inside User Directory

### ISSUE THREAD

**Issue by freMea** [shanalikhan/code-settings-sync#340]
**Visual Studio Code Version :** 1.13.0
**Code Settings Sync Version :** 2.8.1
**Operating System :** Windows 10 x64
**Occurs On:** Help Configuring
**Proxy Enabled:**  No
**GIst Id:** 620405a7199133acf800152aee836f40

Hi,

In my user settings directory, I added two directory to store some custom settings used by VSCode:

- Color Palettes
   - freMea.palette.yml
- Emmet Settings
   - snippets.json

The **_Emmet Settings_** folder and its content **IS** uploaded but **_Color Palettes_** folder is **NOT**. Why? Their files are not empty.

I can't see nothing suspect in _syncLocalSettings.json_ (default one-untouched) or in _settings.json_ or in console.

What I already tried: 

Restart VSCode.

I delete my gist to upload a new fresh sync content in a new gist.

I create another new folder such as:

- test
   - test.txt (not empty)

It is not sync/uploaded

Here is _syncSummary.txt_ extract

```
Files Upload.

extensions.json
snippets.json > Emmet Settings|snippets.json
keybindings.json
locale.json
material-icon-theme.json
projects.json
settings.json
css.json > snippets|css.json
```

Any idea why only **_Emmet Settings_** folder is uploaded?

**Comment by shanalikhan** [comment]
is there any developer console error logged ?

**Comment by freMea** [comment]
Nope.

Did your try to reproduce the steps by creating a dummy folder in your `%APPDATA%\Code\User` directory, add a file to it and upload/update sync?

**Comment by shanalikhan** [comment]
Actually it currently only supports `json` files to upload download in the folders.
I will take that as feature request that allow multiple file types to upload
Like `json`, `yml` , `txt`

**Comment by Cologler** [comment]
Some file do not contains extension name.

**Comment by Cologler** [comment]
Is possible to sync with the file `LICENSE` ?

![picpick-20170913-074623-001](https://user-images.githubusercontent.com/10906962/30353354-f00689d8-9857-11e7-8146-eb529cbedfe1.png)

**Comment by shanalikhan** [comment]
**PR IS WELCOME**

https://github.com/shanalikhan/code-settings-sync/blob/master/src/fileManager.ts#L114
Here i'm getting only JSON files at the moment, you need to pass the list of file types.

Passing from here : https://github.com/shanalikhan/code-settings-sync/blob/e2429a3cb3290a3f2c46e6951bc2a3f96420dcf3/src/extension.ts#L115

Defining configuration list options with `xml,...` file types here :

https://github.com/shanalikhan/code-settings-sync/blob/master/src/setting.ts#L43

**Comment by shanalikhan** [comment]
Released with v2.9
Let me know if there are issues
